opta/module_processors/aws_cloudfront_distribution.py

```diff
@@ -0,0 +1,71 @@
+from typing import TYPE_CHECKING
+
+from opta.exceptions import UserErrors
+from opta.module_processors.base import ModuleProcessor
+from opta.utils import logger
+
+if TYPE_CHECKING:
+    from opta.layer import Layer
+    from opta.module import Module
+
+
+class AwsCloudfrontDstributionProcessor(ModuleProcessor):
+    def __init__(self, module: "Module", layer: "Layer"):
+        if (module.aliased_type or module.type) != "aws-cloudfront-distribution":
+            raise Exception(
+                f"The module {module.name} was expected to be of type aaws cloudfront distribution"
+            )
+        super(AwsCloudfrontDstributionProcessor, self).__init__(module, layer)
+
+    def process(self, module_idx: int) -> None:
+        aws_base_modules = self.layer.get_module_by_type("aws-base", module_idx)
+        from_parent = False
+        if len(aws_base_modules) == 0 and self.layer.parent is not None:
+            from_parent = True
+            aws_base_modules = self.layer.parent.get_module_by_type("aws-base")
+
+        if len(aws_base_modules) == 0:
+            logger.warn(
```

Comment:
We should not output this warning. It can be minor note in the docs but shouldn't be printed at runtime.